src/nunavut/lang/c/templates/_serde_functions.j2

```diff
@@ -0,0 +1,58 @@
+/**
+ * Serialization of {{ composite_type | full_reference_name }} instance into a byte array.
+ *
+ * @param in_instance   A structure instance to serialize.
+ * @param offset        Offset (in bits) of data type in message.
+ * @param out_buffer    Output message buffer.
+ *
+ * @returns             Size of encoded data (in bits)
+ */
+static inline NunavutSignedBitLength {{ composite_type | full_reference_name }}_serialize({{ composite_type | full_reference_name }}* in_instance, NunavutSignedBitLength offset, uint8_t* out_buffer)
+{
+    NunavutSignedBitLength length = 0;
+    NunavutSignedBitLength rc = 0;
+
+    if (out_buffer == NULL)
+    {
+        return NUNAVUT_ERR_INVALID_BUF;
+    }
+    if (offset < 0)
+    {
+        return NUNAVUT_ERR_INVALID_OFFSET;
+    }
+
+    {# TODO fix formatting here #}
+    {%- from '_serialize.j2' import serialize %}
+    {{ serialize(composite_type) }}
+
+    return length;
+}
+
+/**
+ * Deserialization of byte-array encoding into {{ composite_type | full_reference_name }} instance.
+ *
+ * @param out_instance      Structure instance to write data to.
+ * @param offset            Bit offset of structure in data buffer.
+ * @param in_buffer         Message buffer to deserialize.
+ * @param buf_size          Length of input buffer (in bytes) to avoid overflow.
```

Comment:
It's `buf_len` in the code. I suggest using `buf_size_bytes` to make it perfectly unambiguous.